Sources/SwiftDocC/Model/Rendering/Variants/VariantCollection+Coding.swift

```diff
@@ -39,15 +39,19 @@ extension KeyedEncodingContainer {
         )
     }
     
-    /// Encodes the given variant collection.
+    /// Encodes the given variant collection for its non-empty values.
     mutating func encodeVariantCollectionIfNotEmpty<Value>(
         _ variantCollection: VariantCollection<Value>,
         forKey key: Key,
         encoder: Encoder
     ) throws where Value: Collection {
         try encodeIfNotEmpty(variantCollection.defaultValue, forKey: key)
         
-        variantCollection.addVariantsToEncoder(
+        variantCollection.mapValues { value in
+            // Encode `nil` if the value is empty, so that when the patch is applied, it effectively
+            // removes the default value.
+            value.isEmpty ? nil : value
```

Comment:
It will apply every time we call `encodeVariantCollectionIfNotEmpty` in the encoder. 

> I ran into bugs where I would end up with an empty topic section instead of a removed one a few times so making this happen at the encoding level makes a ton of sense to me.

Yes, this is exactly what this code is fixing :)